### ISSUE TITLE
Writing to two buffers in pixel shader doesn't work.

### ISSUE THREAD

**Issue by NETools** [cpt-max/ShaderTest#5]
Consider following hlsl code:

```
float4 MainPS(VertexShaderOutput input) : SV_TARGET
{
     
    uint id = input.Id;
    
    uint lane = id >> 5;
    uint index = id % 32;
	
    InterlockedOr(counterBuffer[lane], 1 << index);
    depthBuffer[0] = 0;

    
    float4 a[3] =
    {
        float4(1, 0, 0, 1),
		float4(0, 1, 0, 1),
		float4(0, 0, 1, 1)
    };
	
 
    return a[id];

}
```

When trying to get the data from counter buffer i am left with bunch of zeroes. But when removing following line:
`depthBuffer[0] = 0;`
i am receiving correct values. Can this be related to https://community.monogame.net/t/calling-dispatchcompute-deletes-data/16967/7?

When writing to two buffers in a pixel shader the value somehow doesnt get passed. Is there a way to fix it?

Best regards

**Comment by cpt-max** [comment]
Writing to buffers from pixel shaders hasn't been tested a lot yet. I'm pretty sure this is a bug. Do you have a simple test project to share that shows the bug?

**Comment by NETools** [comment]
> Owner

Hello, here you go: https://github.com/NETools/MonogameCptForkBug

Best regards

**Comment by cpt-max** [comment]
Thanks a lot. I hope I can take a look at it on the weekend

**Comment by cpt-max** [comment]
It should be fixed in the new 3.8.2.3 nuget
`<PackageReference Include="MonoGame.Framework.Compute.WindowsDX" Version="3.8.2.3" />`

**Comment by NETools** [comment]
Nice! Will' try it out.


Best regards (-:
